src/Symfony/Component/Messenger/HandleTrait.php

```diff
@@ -12,21 +12,27 @@
 namespace Symfony\Component\Messenger;
 
 use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Exception\MultipleHandlersForMessageException;
+use Symfony\Component\Messenger\Exception\NoHandlerForMessageException;
 use Symfony\Component\Messenger\Stamp\HandledStamp;
+use Symfony\Component\Messenger\Stamp\SentStamp;
 
 /**
- * Leverages a message bus to expect a single, synchronous message handling and return its result.
+ * Leverages a message bus to expect a single, synchronous message handling
+ * and return its result; or return null if asynchronous handling.
  *
  * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
  */
 trait HandleTrait
 {
     private MessageBusInterface $messageBus;
+    private bool $allowAsyncHandling = false;
```

Comment:
In my opinion, a new trait for this specific case would be better. You are trying to mix solutions for senders into a trait dedicated to handlers.